bottle.py

```diff
@@ -3976,7 +3989,10 @@ def __init__(self,
         self.name = name
         self.source = source.read() if hasattr(source, 'read') else source
         self.filename = source.filename if hasattr(source, 'filename') else None
-        self.lookup = [os.path.abspath(x) for x in lookup] if lookup else []
+        if lookup:
```

Comment:
Same here, just call `str` on each value unconditionally. The check for an empty lookup variable could also be skipped, as a list comprehension over an empty iterator results in an empty list. This is not a hot path, so a little bit of overhead is fine, readability is more important.

Reply:
Unfortunately, the default value of `lookup` is `None`, because of the common pitfall of Python's mutable types as default parameters. So we have to check if lookup is a falsy value here.